### ISSUE TITLE
v2.0.1 not working in GovCloud region

### ISSUE THREAD

**Issue by timwelch** [udondan/cdk-ec2-key-pair#24]
It looks like something in the EC2 Key Pair MAnager Policy is not getting set correctly for the "partition" .. It's defaulting to the aws commercial instead of pulling the aws-gov or whatever the partition label should be... Is this something in this code base or should I escalate this to the AWS CDK team?

```
17:30:13   0/8 | 10:30:11 PM | UPDATE_FAILED        | AWS::IAM::ManagedPolicy                   | EC2-Key-Pair-Manager-Policy (EC2KeyPairManagerPolicyEBBC1576) Partition "aws" is not valid for resource "arn:aws:ec2:*:*:key-pair/*". (Service: AmazonIdentityManagement; Status Code: 400; Error Code: MalformedPolicyDocument; Request ID: 5aea5c20-43e0-4ec5-9bcf-a9f8f92fde99; Proxy: null)
17:30:13  	new ManagedPolicy (/tmp/jsii-kernel-9ZDg2k/node_modules/@aws-cdk/aws-iam/lib/managed-policy.js:35:26)
17:30:13  	\_ KeyPair.ensureLambda (/tmp/jsii-kernel-9ZDg2k/node_modules/cdk-ec2-key-pair/lib/index.js:125:24)
17:30:13  	\_ new KeyPair (/tmp/jsii-kernel-9ZDg2k/node_modules/cdk-ec2-key-pair/lib/index.js:74:28)
17:30:13  	\_ /usr/local/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:3621:49
17:30:13  	\_ Kernel._wrapSandboxCode (/usr/local/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:4105:16)
17:30:13  	\_ Kernel._create (/usr/local/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:3621:26)
17:30:13  	\_ Kernel.create (/usr/local/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:3356:21)
17:30:13  	\_ KernelHost.processRequest (/usr/local/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:13162:28)
17:30:13  	\_ KernelHost.run (/usr/local/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:13100:14)
17:30:13  	\_ Immediate._onImmediate (/usr/local/lib/python3.7/site-packages/jsii/_embedded/jsii/jsii-runtime.js:13103:37)
17:30:13  	\_ processImmediate (internal/timers.js:461:21)
```

I see the cdk.out json cloudformation for that resource does indeed show the lack of partition awareness...

```
   "EC2KeyPairManagerPolicyEBBC1576": {
      "Type": "AWS::IAM::ManagedPolicy",
      "Properties": {
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "ec2:DescribeKeyPairs",
              "Effect": "Allow",
              "Resource": "*"
            },
            {
              "Action": [
                "ec2:CreateKeyPair",
                "ec2:CreateTags"
              ],
              "Condition": {
                "StringLike": {
                  "aws:RequestTag/CreatedByCfnCustomResource": "CFN::Resource::Custom::EC2-Key-Pair"
                }
              },
              "Effect": "Allow",
              "Resource": "arn:aws:ec2:*:*:key-pair/*"
            },
            {
              "Action": [
                "ec2:CreateTags",
                "ec2:DeleteKeyPair",
                "ec2:DeleteTags"
              ],
              "Condition": {
                "StringLike": {
                  "ec2:ResourceTag/CreatedByCfnCustomResource": "CFN::Resource::Custom::EC2-Key-Pair"
                }
              },
              "Effect": "Allow",
              "Resource": "arn:aws:ec2:*:*:key-pair/*"
            },
```

**Comment by timwelch** [comment]
And the cdk diff output showing the resources being added/changed (going from v1.7.4 to v2.0.1)


```
Resources
[-] Custom::EC2-Key-Pair prodopenvpnkeypairEC2KeyPairprodopenvpnkeypairA122D0B1 destroy
[+] Custom::EC2-Key-Pair prod-openvpn-ec2keypair/EC2-Key-Pair-prod-openvpn-ec2keypair prodopenvpnec2keypairEC2KeyPairprodopenvpnec2keypair99DD7DDA 
[~] AWS::IAM::ManagedPolicy EC2-Key-Pair-Manager-Policy EC2KeyPairManagerPolicyEBBC1576 
 └─ [~] PolicyDocument
     └─ [~] .Statement:
         └─ @@ -1,12 +1,35 @@
            [ ] [
            [ ]   {
            [+]     "Action": "ec2:DescribeKeyPairs",
            [+]     "Effect": "Allow",
            [+]     "Resource": "*"
            [+]   },
            [+]   {
            [ ]     "Action": [
            [ ]       "ec2:CreateKeyPair",
            [+]       "ec2:CreateTags"
            [+]     ],
            [+]     "Condition": {
            [+]       "StringLike": {
            [+]         "aws:RequestTag/CreatedByCfnCustomResource": "CFN::Resource::Custom::EC2-Key-Pair"
            [+]       }
            [+]     },
            [+]     "Effect": "Allow",
            [+]     "Resource": "arn:aws:ec2:*:*:key-pair/*"
            [+]   },
            [+]   {
            [+]     "Action": [
            [+]       "ec2:CreateTags",
            [ ]       "ec2:DeleteKeyPair",
            [-]       "ec2:DescribeKeyPairs"
            [+]       "ec2:DeleteTags"
            [ ]     ],
            [+]     "Condition": {
            [+]       "StringLike": {
            [+]         "ec2:ResourceTag/CreatedByCfnCustomResource": "CFN::Resource::Custom::EC2-Key-Pair"
            [+]       }
            [+]     },
            [ ]     "Effect": "Allow",
            [-]     "Resource": "*"
            [+]     "Resource": "arn:aws:ec2:*:*:key-pair/*"
            [ ]   },
            [ ]   {
            [ ]     "Action": "secretsmanager:ListSecrets",
            @@ -20,7 +43,7 @@
            [ ] ],
            [ ] "Condition": {
            [ ]   "StringLike": {
            [-]     "aws:RequestTag/CreatedBy": "CFN::Resource::Custom::EC2-Key-Pair"
            [+]     "aws:RequestTag/CreatedByCfnCustomResource": "CFN::Resource::Custom::EC2-Key-Pair"
            [ ]   }
            [ ] },
            [ ] "Effect": "Allow",
            @@ -32,6 +55,7 @@
            [ ] "secretsmanager:DeleteSecret",
            [ ] "secretsmanager:DescribeSecret",
            [ ] "secretsmanager:GetResourcePolicy",
            [+] "secretsmanager:GetSecretValue",
            [ ] "secretsmanager:ListSecretVersionIds",
            [ ] "secretsmanager:PutResourcePolicy",
            [ ] "secretsmanager:PutSecretValue",
            @@ -42,7 +66,7 @@
            [ ] ],
            [ ] "Condition": {
            [ ]   "StringLike": {
            [-]     "secretsmanager:ResourceTag/CreatedBy": "CFN::Resource::Custom::EC2-Key-Pair"
            [+]     "secretsmanager:ResourceTag/CreatedByCfnCustomResource": "CFN::Resource::Custom::EC2-Key-Pair"
            [ ]   }
            [ ] },
            [ ] "Effect": "Allow",
```

**Comment by udondan** [comment]
Hi Tim,

thanks for the issue. This is something that needs to and can be fixed in the package itself. I need to use the `this.partition` when generating the policy. Not sure though when I will have the time to look into this. Sometime later this week.

> going from v1.7.4 to v2.0.1

Had you seen the [release notes for version 2.0]( https://github.com/udondan/cdk-ec2-key-pair/releases/tag/v2.0.0)? You cannot easily upgrade from v1 to v2. That's why it got its major version increased.

Cheers,
Daniel

**Comment by timwelch** [comment]
That sounds good to me. I won't be patching this production infrastructure until next month, barring any Critical patches needing to be applied.

And yes, I was patching our production environment and upgrading to the new v2.0.1. It's actually a pretty easy upgrade, except for CloudFormation failing to Delete the old keypair resources. I found that renaming the resource is the 'best' option. 

I bake new AMIs monthly and roll them in using CDK and AutoScaling Rolling updates. So changing the EC2 KeyPair is fine, as the servers are going away. Other than the CFN retrying a few times and complaining it can't delete old KeyPair, the CDK deploy still completes fine in this manner. 

Here is all I change for going from v1.7.4 to v2.0.x (I use Python CDK)

1.) I rename the resource fro `keypair` to `ec2keypair` ...

```
        # Create the Key Pair
        self.keypair = KeyPair(
            self,
            self.id_str('ec2keypair'),
            name=self.id_str('ec2keypair'),
            description='KeyPair for the ' + self.app_name + ' stack'
        )
```

2.) I change the AutoScale Group to use the new `.key_pair_name` instead of the old `.name`

```
            key_name=self.keypair.key_pair_name,
```

**Comment by udondan** [comment]
Hi Tim,

the change has been made via #25 and should be released as v2.1.1 in a minute. Please let me know if this now works in Gov Cloud or I missed something.

Cheers,
Daniel
